fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -414,6 +418,7 @@ void memoOptimize(ConnectContext connectContext, Memo memo, TaskContext rootTask
 
         if (!context.getCandidateMvs().isEmpty()
                 && connectContext.getSessionVariable().isEnableMaterializedViewRewrite()) {
+            new SingleTableMvRewriteRule().transform(tree, context);
```

Comment:
because we skip it in rule-based phase, so should reconsider single table rewrite again. I add it here.
I will add more condition to limit the reuse of this rule.